packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -13,12 +13,18 @@ const PACKAGE_FILENAME = "package.json";
 const BABELIGNORE_FILENAME = ".babelignore";
 
 function exists(filename) {
-  const cached = existsCache[filename];
-  if (cached == null) {
-    return existsCache[filename] = fs.existsSync(filename);
-  } else {
-    return cached;
+  if (existsCache[filename] == null) {
+    if (typeof fs.existsSync !== "function") {
```

Comment:
Though it might be nice to have it near the actual call, so in case this gets ever changed or removed it is obvious that the error also needs to be adjusted.